src/azure-cli-telemetry/azure/cli/telemetry/components/records_collection.py

```diff
@@ -66,6 +71,12 @@ def snapshot_and_read(self):
                       onerror=lambda _, p, tr: self._logger.error('Fail to remove file %s', p))
         self._logger.info('Remove directory %s', tmp)
 
+    def _get_threshold_config(self):
+        config = CLIConfig(config_dir=self._config_dir)
+        threshold = config.getint('telemetry', 'push_data_threshold', fallback=24)
```

Comment:
I think "threshold" is not very straightforward. How about something like

```suggestion
        threshold = config.getint('telemetry', 'push_data_max_wait_time', fallback=24)
```

Reply:
How about `push_data_max_period_in_hours`? `time` can be hours, mins or days. I want to make it more clear.

Reply:
Great! This is very good point!